src/service/DiagramService.js

```
@@ -0,0 +1,33 @@
+import axios from 'axios';
+import { runToastNotification } from '../components/toast-notification/ToastNotification.jsx';
+
+// const baseUrl = 'http://ec-acad-elb-a07a79316f54cbbf.elb.eu-west-2.amazonaws.com:3001';
+const baseUrl = 'http://localhost:3001';
+
+const get = (setNodeData) => {
+  axios
+    .get(`${baseUrl}/page-content/diagram/`)
+    .then((res) => {
+      res?.data?.map((elem) => {
+        setNodeData((prevNodeData) => ({ ...prevNodeData, [elem.nodePosition]: { title: elem.title, body: elem.body } }));
+      });
+    })
+    .catch((error) => {
```

Normally it is better to let the errors propagate to the calling code so it can be handled appropriately.   I think the handling you have here is fine but you are mixing in UI concerns with the service layer.  You should still rethrow the error to the client code instead of swallowing it, since if, for example, any further then were chained off of this at the calling site, then they would be called irrespective of whether an error had occurred.